### PR TITLE
[ADD] national number

### PR DESCRIPTION
**Pull request by victor-champonnois**
## Description
related to :
- cooperator : https://github.com/OCA/cooperative/pull/48
- l10n-belgium https://github.com/OCA/l10n-belgium/pull/177 https://github.com/OCA/l10n-belgium/pull/178

## Odoo task (if applicable)
https://gestion.coopiteasy.be/web#id=9449&model=project.task&view_type=form&menu_id=

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

File: investor_wallet_platform_base/models/subscription_request.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def create(self, vals):
         subscription_request.send_new_subscription_request_notification_email(
             is_company=False
         )
+        subscription_request._compute_display_national_number()
```
Comment (remytms) [inline investor_wallet_platform_base/models/subscription_request.py:60]: Are you shure that this compute method should be called manually at creation of the object ?

File: investor_wallet_platform_base/models/subscription_request.py [hunk 2]
```diff
@@ -67,6 +68,7 @@ def create_comp_sub_req(self, vals):
...
```
Comment (remytms) [inline investor_wallet_platform_base/models/subscription_request.py:71]: Idem.

**Review by remytms — APPROVED** [review]

**Comment by remytms** [comment]
@victor-champonnois Squash your commits plz. :)

File: investor_wallet_platform_base/models/subscription_request.py (hunk 2 above)
Comment (victor-champonnois) [reply investor_wallet_platform_base/models/subscription_request.py:71]: I wasn't sure so I retested, and yes I do. If not, the field National Number will not be displayed when go to the sub request form in the backend.

**Comment by victor-champonnois** [comment]
@remytms commits squashed

**Review by carmenbianca — CHANGES_REQUESTED** [review]
So there's a small problem with this PR. I've marked this 'request changes', but I don't have a good proposition. I'm just blocking the PR from being merged until we have a better understanding.

This PR depends on l10n_be_cooperator_national_number. That module lives in `oca/cooperative`. However, this repository (and the client) do not (yet) use `oca/cooperative`; they use `coopiteasy/vertical-cooperative`.

The PR currently circumvents this problem by pulling in a closed PR <https://github.com/OCA/l10n-belgium/pull/178>, which is an identical-but-unmaintained version of what was merged in <https://github.com/OCA/cooperative/pull/56>. This is not a very good long-term solution. It's good enough to make things work for now, but I'm not very confident accepting this PR with that workaround solution.

My (very malleable) proposition is that we leave this PR open until we have changed this repo to use `oca/cooperative`. Subsequently, we can merge this (and be sure to fix test-requirements.txt) without any workarounds.

**Review by carmenbianca — APPROVED** [review]
LGTM, let's merge this.

**Pull request merged**